src/main/java/hudson/plugins/promoted_builds/PromotionTargetAction.java

```diff
@@ -13,25 +13,17 @@
 public class PromotionTargetAction extends InvisibleAction {
     private final String jobName;
     private final int number;
+    private final AbstractProject<?, ?> project;
 
     public PromotionTargetAction(AbstractBuild<?,?> build) {
         jobName = build.getParent().getFullName();
         number = build.getNumber();
+        project = build.getProject();
     }
 
     public AbstractBuild<?,?> resolve() {
-        AbstractProject<?,?> j = Hudson.getInstance().getItemByFullName(jobName, AbstractProject.class);
-        if (j==null)    return null;
-        return j.getBuildByNumber(number);
+        return project.getBuildByNumber(number);
     }
 
-    public AbstractBuild<?,?> resolve(PromotionProcess parent) {
```

Comment:
It causes a binary compatibility issue.
Is there a real need to remove methods? Probably, you could just ignore input parameters.